src/librustc_mir/transform/generator.rs

```diff
@@ -720,15 +720,18 @@ fn compute_layout<'tcx>(
         _ => bug!(),
     };
 
+    let param_env = tcx.param_env(source.def_id());
+
     for (local, decl) in body.local_decls.iter_enumerated() {
         // Ignore locals which are internal or not live
         if !live_locals.contains(local) || decl.internal {
             continue;
         }
+        let decl_ty = tcx.normalize_erasing_regions(param_env, decl.ty);
```

Comment:
Do you think maybe we should be normalizing during typeck? (i.e. have typeck's `to_const` call `.eval(self.param_env)` on the constant it creates)

Check out the commit titled "borrow_check/type_check: normalize `Aggregate` and `Call` operands.", required to fix #70773, from my PR #70452. Maybe we could fix that issue and this one by just normalizing early.

Reply:
I was wondering where else to do this if not here. Since normalization is just a best-effort thing (at least around constants), doing it earlier is fine, as long as the `param_env` at the earlier site has the same bounds as this one here. Since generator expansion happens happens after inlining, I'm not so sure about that. For this specific ICE it should be fine though.